src/Database/Table/GroupedSelection.php

```diff
@@ -201,6 +201,10 @@ protected function loadRefCache()
 		}
 	}
 
+	protected function emptyResultSet($saveCache = TRUE, $deleteRererencedCache = TRUE) {
+		parent::emptyResultSet($saveCache, false);
```

Comment:
Because of the interation in interation. The deleting referencedCache is launched for each inner interation. This leads to increasing number of sent queries (from 3 to 5 in `query.count.phpt`)